src/x509.cpp

```diff
@@ -181,6 +181,17 @@ AsymmetricPublicKey X509Certificate::getPublicKey() const
     return AsymmetricPublicKey(std::move(pubkey));
 }
 
+bool X509Certificate::checkIfCA() const
+{
+    bool result = true;
+    try {
+         _X509_check_ca(_x509.get());
+    } catch (const OpenSSLException& e) {
```

Comment:
Exceptions should only be used in exceptional cases. I would argue that this is not really exceptional. Once you adapted the callChecked as remarked above, you can just forward this call to it.

Reply:
Agreed and corrected